pkg/stub/handler.go

```diff
@@ -372,10 +472,29 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if srcfg.Name != v1alpha1.SamplesResourceName {
 			return nil
 		}
+
+		// pattern is 1) come in with delete timestample, event delete flag false
+		// 2) then after we remove finalizer, comes in with delete timestamp
+		// and event delete flag true
 		if event.Deleted {
-			logrus.Println("SampleResource deleted")
-			h.samplesResource = nil
-			return h.CreateDefaultResourceIfNeeded()
+			// possibly tell poller to stop
+			logrus.Info("A previous delete attempt has been successfully completed")
+			return nil
+		}
+		if srcfg.DeletionTimestamp != nil {
+			if h.NeedsFinalizing(srcfg) {
+				logrus.Println("Initiating finalizer processing for a SampleResource delete attempt")
+				h.RemoveFinalizer(srcfg)
+				err := h.GoodConditionUpdate(srcfg, corev1.ConditionFalse, v1alpha1.SamplesExist)
+				h.samplesResource = nil
+				timer := time.NewTimer(15 * time.Second)
+				go func() {
+					<-timer.C
```

Comment:
what's the purpose of the 15s delay?

Reply:
conservative delay

would you prefer we launch immediately?  Or some small interval?

Reply:
I don't see any reason not to launch immediately...you were just looking to avoid unnecessary polling iterations i guess?